### ISSUE TITLE
How to retrieve reccuring transactions history?

### ISSUE THREAD

**Issue by DodoTrip** [VivaPayments/API#2270]
Hello, my case is the following:

1) I have a bunch of transactions with their transaction id and an order code that are a result of a reccured payment, passed to me from a migration 
2) I do not know if the transaction id provided is the original

so what i want is:

1) find the original transaction id 
2) get the history that is related to the original transaction id (meaning i need every single transaction after the first one)

TLDR basicly I need to go from a subsequent transaction , trace the original and get the history.

**Comment by VivaWalletCustomerSupport** [comment]
Hello @DodoTrip,

Thank you very much for contacting us. To complete a recurring payment through our [API](https://developer.vivawallet.com/tutorials/payments/create-a-recurring-payment/#step-3-create-a-new-payment-from-the-initial-payment), the transaction id from the original transaction is always needed. Also, all recurring payments together with the initial one have the same order code. I would suggest that you perform a transaction search in your Viva Wallet account with the order code to locate the original transaction and its transaction id.

Kind regards,
Viva Wallet E-commerce Support Team

**Comment by DodoTrip** [comment]
Hello thank you for the fast reply,

I do not wish to complete a reccuring payment i already have this feature implemented. What i want is to track the history of a "subscription" using the transaction id or the order code from any subsequent subscription payment.

Unfortunately i have access only throught the api to complete my task. Is there a way to do this through the api or could you point me to an API call that will give me information about a transaction by providing an order code?

From what i see a lot of api calls are deprecated... 

For example
https://developer.vivawallet.com/apis-for-payments/payment-api/#tag/Transactions-(Deprecated)/paths/~1api~1transactions~1{id}/get
or
https://developer.vivawallet.com/apis-for-payments/payment-api/#tag/Payments-(Deprecated)/paths/~1api~1orders~1{orderCode}/get

**Comment by VivaWalletCustomerSupport** [comment]
Hello @DodoTrip,

Thank you for your feedback. Unfortunately the functionality of retrieving the history of transactions using the transaction id or the order code of the initial payment is not yet available. 

Currently, development for this is not in our pipeline however we've noted your request for future improvements.

About finding the initial payment's transaction id, you can use our [Retrieve Order API call](https://developer.vivawallet.com/apis-for-payments/payment-api/#tag/Payments-(Deprecated)/paths/~1api~1orders~1{orderCode}/get). As a result you will get in the parameter "ExpirationDate", the expiration date of the order code. The default value of the paymentTimeOut parameter in the [create payment order API call](https://developer.vivawallet.com/apis-for-payments/payment-api/#tag/Payments/paths/~1checkout~1v2~1orders/post) is 1800s (30 minutes). If the default value has been used, the payment expiration date will also be the payment date. With the date of the expiration provided you can use our [Sales Transactions API](https://developer.vivawallet.com/apis-for-payments/payment-api/#tag/File-Request-API/paths/~1dataservices~1v1~1transactions~1exports/post) and [WebHook SaleTransactions API](https://developer.vivawallet.com/apis-for-payments/payment-api/#tag/File-Request-API/paths/~1webhooks~1receive%20(your%20endpoint%20for%20webhooks)/post) in order to generate a file with all the transactions of a specific date. In order to use our Sales Transactions API, credentials that are generated by us should be used. 

The owner of the account should contact us via live chat and ask for the Sales Transactions API credentials.

Kind regards,
Viva Wallet E-commerce Support Team

**Comment by VivaWalletCustomerSupport** [comment]
Hello @DodoTrip,

We just wanted to follow up on the reply we sent yesterday. We were simply wondering if you read our response. Are there any other questions, we may help you with?

In any case this ticket will remain active for at least 24 hours, so that you can come back if you wish.

We hope this information helps, however, should you have any questions, please don't hesitate to reply back here and we would be happy to help out!

Kind regards,
Viva Wallet E-commerce Support Team

**Comment by DodoTrip** [comment]
Yes thank you , i am closing this
